File: 1475_final_prices_with_a_special_discount_in_a_shop.java
class Solution {
    public int[] finalPrices(int[] p) {
        int n = p.length, res[] = new int[n];
        for (int i = 0; i < n; i++) {
            res[i] = p[i];
            for (int j = i + 1; j < n; j++) {
                if (p[j] <= p[i]) {
                    res[i] = p[i] - p[j];
                    break;
                }
            }
        }
        return res;
    }
}

File: 1475_final_prices_with_a_special_discount_in_a_shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& lst) {
        int sz = lst.size();
        vector<int> ans(sz);
        for (int i = 0; i < sz; i++) {
            ans[i] = lst[i];
            for (int j = i + 1; j < sz; j++) {
                if (lst[j] <= lst[i]) {
                    ans[i] = lst[i] - lst[j];
                    break;
                }
            }
        }
        return ans;
    }
};
